flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractFlinkResourceReconciler.java

```diff
@@ -257,7 +259,8 @@ private void updateStatusBeforeFirstDeployment(
      * @return True if spec change reconciliation was executed
      */
     protected abstract boolean reconcileSpecChange(
-            FlinkResourceContext<CR> ctx, Configuration deployConfig) throws Exception;
+            FlinkResourceContext<CR> ctx, Configuration deployConfig, boolean requireScale)
```

Comment:
Let's pass the difftype as an argument instead of the confusing boolean param I think.